pkg/cmd/codespace/ssh.go

```diff
@@ -186,6 +207,24 @@ func (a *App) SSH(ctx context.Context, sshArgs []string, opts sshOptions) (err e
 	}
 }
 
+func shouldGenerateSSHKeys(args []string, opts sshOptions) bool {
+	if opts.profile != "" {
+		// The profile may specify the identity file so cautiously don't override that option
+		return false
+	}
+
+	for _, arg := range args {
+		if arg == "-i" {
+			// User specified the identity file so it should exist
+			return false
+		}
+
+		// TODO: should -F have similar behavior?
```

Comment:
Yes I think so, as `-F` would point to a config file that could have a `Host *` stanza that specifies an `IdentityFile`

https://linuxize.com/post/using-the-ssh-config-file/
https://manpages.org/ssh